pkg/reconciler/route/route.go

```diff
@@ -324,33 +324,36 @@ func (c *Reconciler) reconcile(ctx context.Context, r *v1alpha1.Route) error {
 
 	tls := []netv1alpha1.ClusterIngressTLS{}
 	if config.FromContext(ctx).Network.AutoTLS && !resources.IsClusterLocal(r) {
-		allDomains, err := domains.GetAllDomains(ctx, r, getTrafficNames(traffic.Targets))
+		allDomainTagMap, err := domains.GetAllDomainsAndTags(ctx, r, getTrafficNames(traffic.Targets))
 		if err != nil {
 			return err
 		}
-		desiredCert := resources.MakeCertificate(r, allDomains)
-		cert, err := c.reconcileCertificate(ctx, r, desiredCert)
-		if err != nil {
-			r.Status.MarkCertificateProvisionFailed(desiredCert.Name)
-			return err
-		}
+		desiredCerts := resources.MakeCertificates(r, allDomainTagMap)
+		for _, desiredCert := range desiredCerts {
 
-		if cert.Status.IsReady() {
-			r.Status.MarkCertificateReady(cert.Name)
-			r.Status.URL.Scheme = "https"
-			// TODO: we should only mark https for the public visible targets when
-			// we are able to configure visibility per target.
-			setTargetsScheme(&r.Status, "https")
-		} else {
-			r.Status.MarkCertificateNotReady(cert.Name)
-			r.Status.URL = &apis.URL{
-				Scheme: "http",
-				Host:   host,
+			cert, err := c.reconcileCertificate(ctx, r, desiredCert)
+			if err != nil {
+				r.Status.MarkCertificateProvisionFailed(desiredCert.Name)
+				return err
 			}
-			setTargetsScheme(&r.Status, "http")
-		}
 
-		tls = append(tls, resources.MakeClusterIngressTLS(cert, allDomains))
+			if cert.Status.IsReady() {
+				r.Status.MarkCertificateReady(cert.Name)
+				r.Status.URL.Scheme = "https"
```

Comment:
`r.Status.URL.Scheme` corresponds the scheme of the major Route. So we should set it to `https` only when the certificate of the major Route becomes ready.

Reply:
Ok, I see. Do you propose a potential change such as

```suggestion
				if cert.Spec.DNSNames "contains" r.Status.URL.Host {
				     r.Status.URL.Scheme = "https"
				}
```

Would this mean that below, on line 348, we also need to check if the cert is for the major route before setting URL to HTTP?

Reply:
Yes exactly. we also need this on line 348.